source/time_tables/tests/departure_boards_test.cpp

```diff
@@ -205,6 +211,108 @@ BOOST_AUTO_TEST_CASE(departureboard_test1) {
     BOOST_REQUIRE_EQUAL(resp.stop_schedules(1).stop_point().uri(),"stop1");
     BOOST_REQUIRE_EQUAL(resp.stop_schedules(1).stop_point().impact_uris_size(),0);
     }
+
+    // fisrt and last date time tests
+    {
+        // Case 1 :
+        //
+        // Note
+        // - Classical test with a closing date time the day after.
+        //
+        // Input
+        // - opening date time : 05:30:00
+        // - closing date time : 01:30:00
+        // - request date time : 09:45:00
+        //
+        // Output :
+        // - route 1, first date time : 06:00
+        // - route 1, last date time  : 10:00
+        // - route 2, first date time : none
+        // - route 2, last date time  : none
+        b.data->pt_data->routes.front()->line->opening_time = boost::posix_time::duration_from_string("05:30:00.000");
+        b.data->pt_data->routes.front()->line->closing_time = boost::posix_time::duration_from_string("01:30:00.000");
+
+        auto * data_ptr = b.data.get();
+        navitia::PbCreator pb_creator1(data_ptr, bt::second_clock::universal_time(), null_time_period);
+        departure_board(pb_creator1, "stop_point.uri=stop1", {}, {}, d("20150615T094500"), 43200, 0,
+                        10, 0, nt::RTLevel::Base, std::numeric_limits<size_t>::max());
+
+        resp = pb_creator1.get_response();
+        BOOST_REQUIRE_EQUAL(resp.stop_schedules_size(), 2);
+        BOOST_REQUIRE_EQUAL(resp.stop_schedules(0).has_first_datetime(),true);
+        BOOST_REQUIRE_EQUAL(resp.stop_schedules(0).has_last_datetime(), true);
+        BOOST_REQUIRE_EQUAL(resp.stop_schedules(0).first_datetime().time(), "06:00"_t);
```

Comment:
Isn't that test redundant? Better testing whole dt than just time, no?
Same for all those tests